ci/test/interrupt

```diff
@@ -146,63 +146,63 @@ rm running_procs
 # TODO: uncomment to reproduce #587 and #588
 # {
 #
-#mkdir -p actions/sleep
-#
-#cat <<EOF > main.workflow
-#workflow "test" {
-#  resolves = ["sleep"]
-#}
-#
-#action "sleep1" {
-#  uses = "./actions/sleep"
-#  args = "360"
-#}
-#
-#action "sleep2" {
-#  uses = "./actions/sleep"
-#  args = "360"
-#}
-#
-#action "sleep3" {
-#  uses = "./actions/sleep"
-#  args = "360"
-#}
-#
-#action "sleep4" {
-#  uses = "./actions/sleep"
-#  args = "360"
-#}
-#
-#action "sleep5" {
-#  uses = "./actions/sleep"
-#  args = "360"
-#}
-#
-#action "sleep" {
-#  uses = "./actions/sleep"
-#  args = "360"
-#  needs = [
-#    "sleep1",
-#    "sleep2",
-#    "sleep3",
-#    "sleep4",
-#    "sleep5"
-#  ]
-#}
-#EOF
-#
-#popper run --parallel &
-#pid=$!
-#
-## give popper some time to start processes
-#sleep 15
-#
-## -2 is for SIGINT or Ctrl-C
-#kill -2 $pid
-#
-#sleep 5
-#
-#ps aux > running_procs
-#! cat running_procs | grep sleep
-#rm running_procs
+mkdir -p actions/sleep
+
+cat <<EOF > main.workflow
+workflow "test" {
+ resolves = ["sleep"]
+}
+
+action "sleep1" {
+ uses = "./actions/sleep"
+ args = "360"
+}
+
+action "sleep2" {
+ uses = "./actions/sleep"
+ args = "360"
+}
+
+action "sleep3" {
+ uses = "./actions/sleep"
+ args = "360"
+}
+
+action "sleep4" {
+ uses = "./actions/sleep"
+ args = "360"
+}
+
+action "sleep5" {
+ uses = "./actions/sleep"
+ args = "360"
+}
+
+action "sleep" {
+ uses = "./actions/sleep"
+ args = "360"
+ needs = [
+   "sleep1",
+   "sleep2",
+   "sleep3",
+   "sleep4",
+   "sleep5"
+ ]
+}
+EOF
+
+popper run --parallel &
+pid=$!
+
+# give popper some time to start processes
+sleep 15
+
+# -2 is for SIGINT or Ctrl-C
+kill -2 $pid
+
+sleep 5
+
+ps aux > running_procs
+! cat running_procs | grep sleep
```

Comment:
@JayjeetAtGithub I tested on my machine and, while the test does not fail, the behavior is still the same. Not sure why the test is not failing, but this should make it fail accordingly:

```bash
test $(cat running_procs | grep sleep | wc -l) -eq 0
```

Reply:
@ivotron Can you please tell me what behaviour you are talking about ?

Reply:
yes, sorry, I should have explained better. The test has to fail, but it is not failing because line 206 is not catching the error. You can look at the log on travis, specifically [here](https://travis-ci.org/systemslab/popper/jobs/518407405#L1694). There should not be `sleep` processes running. This also fails when `--parallel` is not given, so it seems to be an issue with the way we're killing `HostRunner` processes

Reply:
if you replace line 206 with what I pasted above, the test should fail on your machine

Reply:
Thanks @ivotron ! Let me see if can find out whats going wrong.